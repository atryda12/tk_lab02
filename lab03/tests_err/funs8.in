
int a = 0, b = 0, c = 0;

int gcd(int m, int n) {

int res = 0;
if (m!=n) { 
    if (m > n)
        res = gcd(m-n);
    else
        res = gcd(n-m, m, m);
}
else
    res = m;

return res;
}

int lcm(int a, int b) {
    string s1="",s2="";
    return (a*b) / gcd(s1,s2);
}

int fact(int n) {
  string s = "";
  if(n==0) return s;
  else if(n==1) return 1.0;
  else return n*fact(n-1);
}

int trib(int n) {
    return nbonacci(n,3);
}

int trib(int n) {
    if(n <= 1) {
        return 0;
    }
    if(n == 2) {
        return 1;
    }
    else {
        return trib(n-1) + trib(n-2) + trib(n-3);
    }
}

int pow(int n, int m) {
        int i=1;
        int pow = 1;
        while(i<=m) {
            res = pow*n;
            i = i+1;
        }
        return pow;
}

int gcd_iter(int a, int b) {
    int c = 0;

    while( a != 0 ) {
        c = a;
        a = b % a;
        b = c;
    }
}

print gcd(125,100);
print gcd2(125,100);
print lcm(125,100);


print pow(2,10);

