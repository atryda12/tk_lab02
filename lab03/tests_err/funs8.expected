
Error: Improper number of args in gcd call: line 9
Error: Improper number of args in gcd call: line 11
Error: Improper type of args in gcd call: line 21
Error: Improper returned type, expected int, got string: line 26
Error: Improper returned type, expected int, got string: line 27
Error: Call of undefined fun: 'nbonacci': line 32
Error: Redefinition of function 'trib': line 35
Error: Function identifier 'pow' used as a variable: line 49
Error: Variable 'res' undefined in current scope: line 51
Error: Missing return statement in function 'gcd_iter' returning int: line 65
